feature_engine/scaling/mean_normalization.py

```diff
@@ -0,0 +1,186 @@
+# Authors: Vasco Schiavo <vasco.schiavo@protonmail.com>
+# License: BSD 3 clause
+
+from typing import List, Optional, Union
+
+import pandas as pd
+
+from feature_engine._base_transformers.base_numerical import BaseNumericalTransformer
+from feature_engine._check_init_parameters.check_variables import (
+    _check_variables_input_value,
+)
+from feature_engine._docstrings.fit_attributes import (
+    _feature_names_in_docstring,
+    _n_features_in_docstring,
+    _variables_attribute_docstring,
+)
+from feature_engine._docstrings.init_parameters.all_trasnformers import (
+    _variables_numerical_docstring,
+)
+from feature_engine._docstrings.methods import (
+    _fit_not_learn_docstring,
+    _fit_transform_docstring,
+    _inverse_transform_docstring,
+)
+from feature_engine._docstrings.substitute import Substitution
+
+
+@Substitution(
+    variables=_variables_numerical_docstring,
+    variables_=_variables_attribute_docstring,
+    feature_names_in_=_feature_names_in_docstring,
+    n_features_in_=_n_features_in_docstring,
+    fit=_fit_not_learn_docstring,
+    fit_transform=_fit_transform_docstring,
+    inverse_transform=_inverse_transform_docstring,
+)
+class MeanNormalizationScaler(BaseNumericalTransformer):
+    """
+    The MeanNormalizationScaler() applies the mean normalization scaling techinques
+    to one or multuple columns of a dataframe.
+
+    Mean normalization is a way to implement feature scaling. Mean normalization
+    calculates and subtracts the mean of for every feature and divide this value
+    by the range (max - min).
+
+    A list of variables can be passed as an argument. Alternatively, the transformer
+    will automatically select and transform all variables of type numeric.
+
+    If a column is constant, then it will be transformed to the zero constant column.
+
```

Comment:
here we need to add a sentence that says more details in the user guide and create a user guide file with more detail about this transformation and an example with a real dataframe.

Here we create the user guide files: https://github.com/feature-engine/feature_engine/tree/main/docs/user_guide

We need a new folder for scaling, there an index, and then the file for the meannormalization scaler.

And also, to build the docs, we need to add a file to the api: https://github.com/feature-engine/feature_engine/tree/main/docs/api_doc  here we need a folder for scaling and in that folder an index, and the file for the scaler. 

If you check out the folder transformation inside those folders, you can very quickly get an idea of the files that we need and their format. The one for the api is super straightforward, the one for the user guide needs more work but I can support.